internal/proxy/proxy.go

```diff
@@ -58,10 +61,20 @@ type Config struct {
 
 func (c Config) DialerOpts() []cloudsqlconn.Option {
 	var opts []cloudsqlconn.Option
-	if c.Token != "" {
+	// The ordering in the switch statement is deliberate. When multiple auth
+	// values are configured, the precedence is:
+	// 1. token
+	// 2. credentials file
+	// 3. gcloud auth
+	switch {
+	case c.Token != "":
 		opts = append(opts, cloudsqlconn.WithTokenSource(
 			oauth2.StaticTokenSource(&oauth2.Token{AccessToken: c.Token}),
 		))
+	case c.CredentialsFile != "":
+		opts = append(opts, cloudsqlconn.WithCredentialsFile(
+			c.CredentialsFile,
+		))
```

Comment:
I think we should throw an error if the user specifies both. Silently ignoring one makes it harder to understand what's wrong.

Reply:
It would also be good if we can log which one we are using on start.

Reply:
I've put logging here, and moves the check for both values up into `parseConfig` -- that's where other argument checking is happening.